diff --git a/src/extract_gpuinfo_amdgpu.c b/src/extract_gpuinfo_amdgpu.c
index b5d4584..b5a974e 100644
--- a/src/extract_gpuinfo_amdgpu.c
+++ b/src/extract_gpuinfo_amdgpu.c
@@ -158,7 +158,7 @@ struct gpu_vendor gpu_vendor_amdgpu = {
 
 static int readAttributeFromDevice(nvtop_device *dev, const char *sysAttr, const char *format, ...);
 
-__attribute__((constructor)) static void init_extract_gpuinfo_amdgpu(void) { register_gpu_vendor(&gpu_vendor_amdgpu); }
+void init_extract_gpuinfo_amdgpu(void) { register_gpu_vendor(&gpu_vendor_amdgpu); }
 
 static int wrap_drmGetDevices(drmDevicePtr devices[], int max_devices) {
   assert(_drmGetDevices2 || _drmGetDevices);
diff --git a/src/extract_gpuinfo_intel.c b/src/extract_gpuinfo_intel.c
index a57f9fb..9404d23 100644
--- a/src/extract_gpuinfo_intel.c
+++ b/src/extract_gpuinfo_intel.c
@@ -100,7 +100,7 @@ static struct gpu_info_intel *gpu_infos;
 // Discrete GPU are others
 #define INTEGRATED_I915_GPU_PCI_ID "0000:00:02.0"
 
-__attribute__((constructor)) static void init_extract_gpuinfo_intel(void) { register_gpu_vendor(&gpu_vendor_intel); }
+void init_extract_gpuinfo_intel(void) { register_gpu_vendor(&gpu_vendor_intel); }
 
 bool gpuinfo_intel_init(void) { return true; }
 void gpuinfo_intel_shutdown(void) {
diff --git a/src/extract_gpuinfo_msm.c b/src/extract_gpuinfo_msm.c
index 244240e..7cf46f6 100644
--- a/src/extract_gpuinfo_msm.c
+++ b/src/extract_gpuinfo_msm.c
@@ -146,7 +146,7 @@ static void authenticate_drm(int fd) {
 #define STRINGIFY(x) STRINGIFY_HELPER_(x)
 #define STRINGIFY_HELPER_(x) #x
 
-__attribute__((constructor)) static void init_extract_gpuinfo_msm(void) { register_gpu_vendor(&gpu_vendor_msm); }
+void init_extract_gpuinfo_msm(void) { register_gpu_vendor(&gpu_vendor_msm); }
 
 bool gpuinfo_msm_init(void) {
   libdrm_handle = dlopen("libdrm.so", RTLD_LAZY);
diff --git a/src/extract_gpuinfo_nvidia.c b/src/extract_gpuinfo_nvidia.c
index f0356ed..4e6baca 100644
--- a/src/extract_gpuinfo_nvidia.c
+++ b/src/extract_gpuinfo_nvidia.c
@@ -201,7 +201,7 @@ struct gpu_vendor gpu_vendor_nvidia = {
     .name = "NVIDIA",
 };
 
-__attribute__((constructor)) static void init_extract_gpuinfo_nvidia(void) { register_gpu_vendor(&gpu_vendor_nvidia); }
+void init_extract_gpuinfo_nvidia(void) { register_gpu_vendor(&gpu_vendor_nvidia); }
 
 /*
  *
